src/ATen/native/xpu/sycl/Indexing.cpp

```diff
@@ -1550,11 +1550,11 @@ struct IndexSelectSparse3Functor {
       index_t count,
       index_t offset,
       index_t first_match) const {
-    index_t* __restrict__ ptr_res_dim_indices_out =
+    index_t* RESTRICT ptr_res_dim_indices_out =
```

Comment:
`__restrict__` seems like compiler level definition. Does Windows DPC++ compiler build miss this definition? That's strange assuming same version of the compiler used on Windows and Linux.

Who defines `RESTRICT` macro and where?

Reply:
`RESTRICT` is defined by:
https://github.com/intel/torch-xpu-ops/blob/33da0a1464e294c826b1844d21ac74bf99d5e5e3/src/comm/Macros.h#L3-L7
We should use `__restrict` for MSVC.